scripts/install_submitter.py

```
@@ -0,0 +1,619 @@
+Handles installation, environment setup, and VRED preferences configuration.
+
+PREREQUISITES AND ASSUMPTIONS:
+
+- System Requirements:
+  - Windows OS: VRED Pro is Windows-only, other OS will be rejected
+  - Python 3.x: Required for script execution
+  - Administrator privileges: Needed for registry modification and system environment variables
+
+- VRED PRO Installation:
+  - VRED Pro 2025/2026: At least one version must be installed at:
+    - C:/Program Files/Autodesk/VREDPro-17.3 (2025)
+    - C:/Program Files/Autodesk/VREDPro-18.0 (2026)
+  - Site-packages directory: [VRED_INSTALL]/lib/python/Lib/site-packages/ must exist
+
+- Dependency Bundle:
+  - If dependency_bundle/ directory is empty or missing:
+    - Internet connection required for pip install commands
+    - PyPI access needed to download packages (deadline, xxhash, psutil, etc.)
+  - If dependency_bundle/*.zip files exist: Offline installation possible
+
+- RUNTIME ASSUMPTIONS:
+  - Single-user installation (not system-wide)
+  - VRED should be closed during installation for best results
+"""
+
+import argparse
+import base64
+import binascii
+import ctypes
+import logging
+import os
+import platform
+import shutil
+import sys
+import xml.etree.ElementTree as ET
+import zipfile
+from pathlib import Path
+from typing import List, Optional, TYPE_CHECKING
+
+# Import dependency bundle builder
+# Add the scripts directory to Python path for importing deps_bundle
+project_scripts_dir = Path(__file__).parent
+if str(project_scripts_dir) not in sys.path:
+    sys.path.insert(0, str(project_scripts_dir))
+
+try:
+    from deps_bundle import build_deps_bundle
+except ImportError as e:
+    print(f"Warning: Could not import deps_bundle: {e}")
+    build_deps_bundle = None
+
+if TYPE_CHECKING or platform.system().lower() == "windows":
+    try:
+        import winreg
+    except ImportError:
+        winreg = None  # type: ignore
```

Is it ok that we set this to `none` on Windows? Why wouldn't we just fail fast here? If there's a reason we should probably leave a comment. Same with the budle_deps_bundle import above

---

I agree, we should fail fast rather than silently setting to None. I've updated the code to raise an exception.